Xamarin.Forms.Core/NameScopeExtensions.cs

```diff
@@ -1,11 +1,21 @@
+using System;
 using Xamarin.Forms.Internals;
 
 namespace Xamarin.Forms
 {
 	public static class NameScopeExtensions
 	{
 		public static T FindByName<T>(this Element element, string name)
-		=> (T)element.FindByName(name);
+		{
+			try {
+				return (T)element.FindByName(name);
+			}
+			catch (InvalidCastException) {
+				if (DesignMode.IsDesignModeEnabled)
```

Comment:
`DesignMode` doesn't mean the previewer is plugged in

Reply:
Do you mean... `IsDesignModeEnabled` could be true without the previewer, or `IsDesignModeEnabled` could be false with the previewer?

It is only set if `XamlFileProvider` or `ResourceProvider2` are set. Are there non-previewer scenarios these are used?